### PR TITLE
Return error when library not found

### PR DESCRIPTION
**Pull request by LambdaHJ**
Signed-off-by: 黄金 <huangjin_henan@outlook.com>

Init function return an error when library open failed.

### CODE REVIEW

File: pkg/nvml/init.go
Comment (elezar) [inline pkg/nvml/init.go:40]: Although this is cleaner than a `panic` from certain perspectives this is not a documented error returned by the underlying `nvmlInit` function.

What we tend to do internally is use a wrapper (e.g. `tryNvmlInit`) that recovers from the panic and converts the `Return` to an `error` instead.

With that said, I'll let @klueska weigh in on this change.

Comment (klueska) [reply pkg/nvml/init.go:40]: This comes up frequently enough as a perceived issue that I'm inclined to go ahead and change it since there is in fact an error called `ERROR_LIBRARY_NOT_FOUND`.

File: pkg/nvml/init.go
Comment (klueska) [inline pkg/nvml/init.go:25]: This is unnecessary, please remove.

**Comment by elezar** [comment]
Thanks for the review @klueska.

@LambdaHJ please note that before we can merge this we would need to make the change in the correct place.

You would need to update the source in the `gen` folder [here](https://github.com/NVIDIA/go-nvml/blob/main/gen/nvml/init.go#L34), regenerate the bindings, and then also add the changes in `pkg` to Git.

To regenerate the bindings running:
```
make docker-bindings
```
should do this.

**Comment by jasonliu747** [comment]
Hi @klueska @elezar , could you make a release after this PR? Really appreciated.

**Comment by fmoessbauer** [comment]
> Hi @klueska @elezar , could you make a release after this PR? Really appreciated.

I just stumbled upon the same issue and was happy to see that there is already an MR. We also need this to test if NVML is available on the system.

**Review by klueska — APPROVED** [review]

**Comment by klueska** [comment]
Looks good. @elezar can you do a quick test to make sure it works as expected before merging.

**Comment by elezar** [comment]
Testing in a docker container on my mac (where I know there is no NVML).

On `main`:
```
$ make examples
$ ./devices
panic: error opening libnvidia-ml.so.1: libnvidia-ml.so.1: cannot open shared object file: No such file or directory

goroutine 1 [running]:
github.com/NVIDIA/go-nvml/pkg/nvml.Init(0xc000012000)
        /Users/elezar/src/go-nvml/pkg/nvml/init.go:41 +0x177
main.main()
        /Users/elezar/src/go-nvml/examples/devices/main.go:27 +0x48
```

With these changes:
```
$ ./devices
./devices: symbol lookup error: ./devices: undefined symbol: nvmlErrorString
```

Which is expected since the test code contains:
```
	ret := nvml.Init()
	if ret != nvml.SUCCESS {
		log.Fatalf("Unable to initialize NVML: %v", nvml.ErrorString(ret))
	}
```

Changing this to
```
	ret := nvml.Init()
	if ret == nvml.ERROR_LIBRARY_NOT_FOUND {
		log.Fatalf("Could not load NVML library")
	} else if ret != nvml.SUCCESS {
		log.Fatalf("Unable to initialize NVML: %v", nvml.ErrorString(ret))
	}
```
We see:
```
$ ./devices
2022/07/15 11:16:55 Could not load NVML library
```

@LambdaHJ one thing I noted is that running `make docker-bindings` shows:
```
diff --git a/pkg/nvml/init.go b/pkg/nvml/init.go
index 962823b..1572f81 100644
--- a/pkg/nvml/init.go
+++ b/pkg/nvml/init.go
@@ -36,6 +36,7 @@ func Init() Return {
        if err != nil {
                return ERROR_LIBRARY_NOT_FOUND
        }
+
        nvml = lib
        updateVersionedSymbols()

```
Could you run that again or add the newline as indicated?

File: pkg/nvml/init.go
Comment (elezar) [inline pkg/nvml/init.go:39]: Please add the newline above this.

**Review by elezar — CHANGES_REQUESTED** [review]

File: pkg/nvml/init.go
Comment (LambdaHJ) [reply pkg/nvml/init.go:40]: done

**Review by elezar — APPROVED** [review]

**Pull request merged**
